cdap-master/src/main/java/co/cask/cdap/data/tools/ReplicationStatusTool.java

```diff
@@ -0,0 +1,379 @@
+
+    //Check all regions are accounted on both clusters
+    if (masterTimeMap.size() != slaveTimeMap.size()) {
+      System.out.println("Number of regions on the Master and Slave Clusters do not match."
+                         + " HBase Replication is NOT complete.");
+      return;
+    }
+
+    //Check the maps for all regions
+    for (Map.Entry<String, Long> writeTimeEntry: masterTimeMap.entrySet()) {
+      String region = writeTimeEntry.getKey();
+      long writeTime = writeTimeEntry.getValue();
+      long replicateTime = slaveTimeMap.get(region);
+
+      if (!isReplicationComplete(masterShutdownTime, writeTime, replicateTime)) {
+        System.out.println("HBase Replication is NOT complete. Region:"
+                             + region + " behind by "
+                             + (writeTime - replicateTime) + "ms.");
+        return;
+      }
+    }
+    // If replication is complete for all regions.
+    System.out.println("HBase Replication is complete.");
+  }
+
+  private static Map<String, Long> readMasterStatusFile(String masterStatusFile) throws IOException {
+    Map<String, Long> map = new HashMap<String, Long>();
+    try (BufferedReader br = new BufferedReader(new FileReader(masterStatusFile))) {
+      String line;
+      //Read the shutdown Time
+      masterShutdownTime = new Long(br.readLine());
+      //Read the checksum
+      masterChecksum = br.readLine();
+
+      //Read the rest into the region map
+      while ((line = br.readLine()) != null) {
+        StringTokenizer st = new StringTokenizer(line, delimiter);
+        String region = st.nextToken();
+        Long timestamp = new Long(st.nextToken());
+        map.put(region, timestamp);
+      }
+    }
+    return map;
+  }
+
+  private static void writeMasterStatusFile(Map<String, Long> map, String filePath) throws IOException {
+    FileWriter fw = new FileWriter(filePath);
+    BufferedWriter bw = new BufferedWriter(fw);
+    //Write shutdown time
+    bw.write(masterShutdownTime.toString());
+    bw.newLine();
+    //Write checksum
+    bw.write(masterChecksum);
+    bw.newLine();
+
+    //Write region map
+    for (Map.Entry<String, Long> entry: map.entrySet()) {
+      bw.write(entry.getKey() + delimiter + entry.getValue());
+      bw.newLine();
+    }
+    bw.close();
+  }
+
+  private static Map<String, Long> getMapFromTable(String qualifier) throws IOException {
+    HBaseTableUtil tableUtil = new HBaseTableUtilFactory(cConf).get();
+    HTable hTable = tableUtil.createHTable(hConf, getReplicationStateTableId(tableUtil));
+
+    // Scan the table to scan for all regions.
+    ScanBuilder scan = tableUtil.buildScan();
+    scan.addColumn(Bytes.toBytes(ReplicationConstants.ReplicationStatusTool.TIME),
+                   Bytes.toBytes(qualifier));
+    Result result;
+    HashMap<String, Long> timeMap = new HashMap<>();
+
+    try (ResultScanner resultScanner = hTable.getScanner(scan.build())) {
```

Comment:
No they can have entries from another cluster ID. If I fail over from cluster 1 to cluster 2, then set up cluster 2 to replicate to cluster 1 so that I can fail back over.